src/components/Footer/Footer.tsx

```diff
@@ -0,0 +1,66 @@
+import React, { FC, Dispatch } from 'react';
+import { Filter } from '../../types/Filter';
+import cn from 'classnames';
+
+type Props = {
+  filterOption: Filter;
+  uncompletedTodosAmount: number;
+  setFilterOption: Dispatch<React.SetStateAction<Filter>>;
+};
+
+export const Footer: FC<Props> = ({
+  filterOption,
+  uncompletedTodosAmount,
+  setFilterOption,
+}) => {
+  return (
+    <footer className="todoapp__footer" data-cy="Footer">
+      <span className="todo-count" data-cy="TodosCounter">
+        {uncompletedTodosAmount} items left
+      </span>
+
+      <nav className="filter" data-cy="Filter">
+        <a
+          href="#/"
+          className={cn('filter__link', {
+            selected: filterOption === Filter.All,
+          })}
+          data-cy="FilterLinkAll"
+          onClick={() => setFilterOption(Filter.All)}
+        >
+          All
+        </a>
```

Comment:
You can convert your enum into an array with `Object.keys/values/entries` and map through it to generate links